test/extended/networking/vlan.go

```diff
@@ -0,0 +1,134 @@
+			exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName1, func(pod *kapiv1.Pod) {
+				vlanAnnotation := fmt.Sprintf("%s/%s@%s", namespace, vlan1, vlan1)
+				pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s", bridgeAnnotation, vlanAnnotation)}
+			})
+			pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), podName1, metav1.GetOptions{})
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			networkStatusString, ok := pod.Annotations["k8s.v1.cni.cncf.io/network-status"]
+			o.Expect(ok).To(o.BeTrue())
+			o.Expect(networkStatusString).ToNot(o.BeNil())
+			networkStatuses := []nadtypes.NetworkStatus{}
+			o.Expect(json.Unmarshal([]byte(networkStatusString), &networkStatuses)).ToNot(o.HaveOccurred())
+			o.Expect(networkStatuses).To(o.HaveLen(3))
+			o.Expect(networkStatuses[2].Interface).To(o.Equal(vlan1))
+			o.Expect(networkStatuses[2].Name).To(o.Equal(fmt.Sprintf("%s/%s", namespace, vlan1)))
+
+			g.By("having a second pod pinging the first pod")
+			exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName2, func(pod *kapiv1.Pod) {
+				vlanAnnotation := fmt.Sprintf("%s/%s@%s", namespace, vlan2, vlan2)
+				pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s", bridgeAnnotation, vlanAnnotation)}
+				pod.Spec.Containers[0].Command = []string{"/bin/bash", "-c", fmt.Sprintf("ping -c 1 %s", ip1)}
+			})
+
+			g.By("having another pod on different vlan unanble to ping the first pod")
+
+			podDefinition := frameworkpod.NewAgnhostPod(f.Namespace.Name, podName3, nil, nil, nil)
+			vlanAnnotation := fmt.Sprintf("%s/%s@%s", namespace, otherVlan, otherVlan)
+			podDefinition.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s", bridgeAnnotation, vlanAnnotation)}
+			podDefinition.Spec.Containers[0].Args = nil
+			podDefinition.Spec.Containers[0].Command = []string{"/bin/bash", "-c", fmt.Sprintf("ping -c 1 %s && touch /tmp/ready", ip1)}
+			podDefinition.Spec.Containers[0].ReadinessProbe = &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					Exec: &corev1.ExecAction{
+						Command: []string{"/bin/bash", "-c", fmt.Sprintf("test -f /tmp/ready")},
+					},
+				},
+			}
+
+			podDefinition.Spec.Containers[0].SecurityContext = podframework.GetRestrictedContainerSecurityContext()
+			podDefinition.Spec.SecurityContext = podframework.GetRestrictedPodSecurityContext()
+			otherVlandPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), podDefinition, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "error creating pod")
+			err = wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+				retrievedPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), otherVlandPod.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				return retrievedPod.Status.ContainerStatuses[0].Ready == false, nil
```

Comment:
there is a panic here

Reply:
STEP: having another pod on different vlan unanble to ping the first pod 02/20/23 22:04:26.144
E0220 22:04:26.297917   58812 runtime.go:79] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 766 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x825abe0?, 0xc0048cf980})
	k8s.io/apimachinery@v0.25.0/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	k8s.io/apimachinery@v0.25.0/pkg/util/runtime/runtime.go:49 +0x75
panic({0x825abe0, 0xc0048cf980})
	runtime/panic.go:884 +0x212
github.com/openshift/origin/test/extended/networking.glob..func14.1.3()
	github.com/openshift/origin/test/extended/networking/vlan.go:129 +0xd5
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x18, 0xc0000d6c00})
	k8s.io/apimachinery@v0.25.0/pkg/util/wait/wait.go:222 +0x1b
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x9462ed8?, 0xc00007e130?}, 0xc004803c28?)
	k8s.io/apimachinery@v0.25.0/pkg/util/wait/wait.go:235 +0x57
k8s.io/apimachinery/pkg/util/wait.poll({0x9462ed8, 0xc00007e130}, 0x68?,

Reply:
The test probably executes before the container is created.
Adding a check for len of containers.